pkg/operator/controller/ingress/internal_service.go

```diff
@@ -137,6 +137,7 @@ func internalServiceChanged(current, expected *corev1.Service) (bool, *corev1.Se
 		cmpopts.IgnoreMapEntries(func(k, _ string) bool {
 			return !managedInternalServiceAnnotations.Has(k)
 		}),
+		cmpopts.EquateEmpty(),
```

Comment:
I removed this and the unit tests still pass. Is this expected? Were the additional unit tests expected to verify `EquateEmpty()`?

Reply:
When I comment this line out, `TestInternalServiceChangedEmptyAnnotations` fails:
```
--- FAIL: TestInternalServiceChangedEmptyAnnotations (0.00s)
    --- FAIL: TestInternalServiceChangedEmptyAnnotations/null_to_empty (0.00s)
        internal_service_test.go:262: 
            	Error Trace:	/home/mmasters/src/github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress/internal_service_test.go:262
            	Error:      	Should be false
            	Test:       	TestInternalServiceChangedEmptyAnnotations/null_to_empty
    --- FAIL: TestInternalServiceChangedEmptyAnnotations/empty_to_null (0.00s)
        internal_service_test.go:262: 
            	Error Trace:	/home/mmasters/src/github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress/internal_service_test.go:262
            	Error:      	Should be false
            	Test:       	TestInternalServiceChangedEmptyAnnotations/empty_to_null
```
This is with Go 1.20.2.